cmd/build_worker_image.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/temporalio/omes/cmd/cmdoptions"
+	"go.uber.org/zap"
+	"golang.org/x/mod/semver"
+)
+
+func buildWorkerImageCmd() *cobra.Command {
+	var b workerImageBuilder
+	cmd := &cobra.Command{
+		Use:   "build-worker-image",
+		Short: "Build an image",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := b.build(cmd.Context()); err != nil {
+				b.logger.Fatal(err)
+			}
+		},
+	}
+	b.addCLIFlags(cmd.Flags())
+	cmd.MarkFlagRequired("language")
+	cmd.MarkFlagRequired("version")
+	return cmd
+}
+
+type workerImageBuilder struct {
+	logger         *zap.SugaredLogger
+	language       string
+	version        string
+	semverTags     string
+	platform       string
+	imageName      string
+	dryRun         bool
+	tags           []string
+	labels         []string
+	loggingOptions cmdoptions.LoggingOptions
+}
+
+func (b *workerImageBuilder) addCLIFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&b.language, "language", "", "Language to build a worker image for")
+	fs.StringVar(&b.version, "version", "",
+		"SDK version to build a worker image for - treated as path if slash present, but must be beneath this dir and at least one tag required")
+	fs.StringVar(&b.semverTags, "semver-tags", "",
+		"Can be 'minor', 'major', or 'all' - affects additional tags, only used on non-path version")
+	fs.StringVar(&b.platform, "platform", "", "Platform for use in docker build --platform")
+	fs.StringVar(&b.imageName, "image-name", "omes", "Name of the image to build")
+	fs.BoolVar(&b.dryRun, "dry-run", false, "If set, just print the commands that would run but do not run them")
+	fs.StringSliceVar(&b.tags, "image-tag", nil, "Additional tags to add to the image")
+	fs.StringSliceVar(&b.labels, "image-label", nil, "Additional labels to add to the image")
+	b.loggingOptions.AddCLIFlags(fs)
+}
+
+func (b *workerImageBuilder) build(ctx context.Context) error {
+	b.logger = b.loggingOptions.MustCreateLogger()
+	lang, err := normalizeLangName(b.language)
+	if err != nil {
+		return err
+	}
+
+	// Setup args, tags, and labels based on version
+	isPathVersion := strings.ContainsAny(b.version, `/\`)
+	var buildArgs []string
+	if isPathVersion {
+		if len(b.tags) == 0 {
+			return fmt.Errorf("at least one tag required for path version")
+		} else if s, err := os.Stat(b.version); err != nil {
+			return fmt.Errorf("invalid path version: %w", err)
+		} else if !s.IsDir() {
+			return fmt.Errorf("invalid path version: must be dir")
+		} else if !filepath.IsLocal(b.version) {
+			return fmt.Errorf("invalid path version: must be beneath this dir")
+		}
+		// We have to copy the entire
```

Comment:
Copy the entire!

Reply:
Yup, copy the entire everything except the entire comment out of my head

Reply:
:+1: Will fix in next PR coming shortly after this one.